src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
-from ..shapes import BufferMeshShape, FileMeshShape, Shape, shape_factory
+from ..bsdfs import LambertianBSDF, OpacityMaskBSDF
+from ..geometry import PlaneParallelGeometry, SceneGeometry, SphericalShellGeometry
+from ..shapes import (
+    BufferMeshShape,
+    FileMeshShape,
+    RectangleShape,
+    Shape,
+    SphereShape,
+    shape_factory,
+)
 from ...attrs import documented, get_doc, parse_docs
+from ...constants import EARTH_RADIUS
 from ...units import to_quantity
 from ...units import unit_context_config as ucc
+from ...units import unit_context_kernel as uck
 from ...units import unit_registry as ureg
 
 
-@parse_docs
-@attrs.define(eq=False, slots=False)
-class DEMSurface(Surface):
+def mesh_from_dem(
+    da: xr.DataArray,
+    geometry: SceneGeometry,
+    planet_radius: pint.Quantity | float = EARTH_RADIUS,
+):
     """
-    DEM surface [``dem``]
+    Construct a DEM surface from a dataarray holding elevation data.
+    There are two ways of specifying the dataarray:
+    - With latitude and longitude based coordinates:
+      The coordinates must be named "lat" and "lon" and be in that order.
+    - With x and y length based coordinates
+      The coordinates must be named "x" and "y" and be in that order.
+
+    The unit of the coordinates must be given by the `coordinate_units` attribute in the DataArray,
+    while the unit of the elevation data must be given by the `unit` attribute for automatic conversion.
 
-    A surface description for digital elevation models.
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray with elevation data, indexed either by latitude and longitude coordinates or x and y lenghts
 
-    This surface supports instantiation based on an xarray data array, a
-    triangulated mesh file in the PLY and OBJ formats and an analytical
-    elevation mapping the x and y coordinates to an elevation value.
+    geometry : SceneGeometry
+        Geometry of the scene.
 
-    .. admonition:: Class method constructors
+    planet_radius : pint.Quantity
+        Planet radius, used for converting between latitude/longitude and x/y coordinates *ONLY* in the case
+        of a plane parallel scene geometry.
 
-       .. autosummary::
+    id : str
+        Identifier of the scene element.
 
-          from_dataarray
-          from_analytical
+    Returns
+    -------
 
-    Notes
-    -----
-    * Contrary to most other surfaces, this scene element expands as a single
-      Mitsuba Shape plugin which includes its child BSDF rather than referencing
-      a top-level dictionary entry. The reason for this is that some allowed
-      shapes expand as Mitsuba instances rather than dictionaries, which
-      prevents object referencing.
+    mesh : :class:`.BufferMeshShape`
+        A triangulated mesh representing the DEM
+
+    theta_lim : tuple
+        The limits of the latitude extent of the DEM
+
+    phi_lim : tuple
+        The limits of the longitude extent of the DEM
+    """
+
```

Comment:
Safer behaviour (you don't want to accidentally mutate `EARTH_RADIUS`):

```suggestion
    if isinstance(geometry, PlaneParallelGeometry):
        if planet_radius is None:
            planet_radius = EARTH_RADIUS

    else:  # if isinstance(geometry, SphericalShellGeometry)
        if planet_radius is not None:
            # Warn user that the planet_radius arg will be ignored
```

Reply:
I am not quite sure, why you would accidentally mutate EARTH_RADIUS in the other case :thinking: